reader/components/enhancement-sidebar/EnhancementSidebar.jsx

```diff
@@ -1,18 +1,36 @@
 import React from 'react'
-import { sortBy } from 'lodash'
 import './EnhancementSidebar.scss'
 import { useGlobalStore } from '../../store'
 
 const EnhancementSidebar = () => {
   const [{ ui, document }] = useGlobalStore()
 
-  const sortedAnnotations = sortBy(document.annotations, [
-    annotation => sortBy(Object.keys(annotation.rects))[0],
-    annotation =>
-      annotation.rects[sortBy(Object.keys(annotation.rects))[0]][0].top,
-    annotation =>
-      annotation.rects[sortBy(Object.keys(annotation.rects))[0]][0].left,
-  ])
+  const sortedAnnotations = Object.keys(document.annotations)
+    .map(annotationId => [annotationId, document.annotations[annotationId]])
+    .sort(([, firstAnnotation], [, secondAnnotation]) => {
+      const firstAnnotationId = Object.keys(firstAnnotation.rects).sort()[0]
+      const secondAnnotationId = Object.keys(secondAnnotation.rects).sort()[0]
+
+      if (firstAnnotationId < secondAnnotationId) return -1
+      if (firstAnnotationId > secondAnnotationId) return 1
+
+      const firstAnnotationsSorted = firstAnnotation.rects[firstAnnotationId]
+        .map(f => [f.top, f.left])
+        .sort()
+      const secondAnnotationsSorted = secondAnnotation.rects[secondAnnotationId]
+        .map(f => [f.top, f.left])
+        .sort()
+
+      if (firstAnnotationsSorted[0][0] < secondAnnotationsSorted[0][0])
+        return -1
+      if (firstAnnotationsSorted[0][0] > secondAnnotationsSorted[0][0]) return 1
+
+      if (firstAnnotationsSorted[0][1] < secondAnnotationsSorted[0][1])
+        return -1
+      if (firstAnnotationsSorted[0][1] > secondAnnotationsSorted[0][1]) return 1
```

Comment:
Basically, you have to return only numeric value. The `sort()` function only checks for greater than 0 or less than 0.

If I understand the algorithm correctly, your algorithm might be cut to the following:

```js
return (
  (firstAnnotationsSorted[0][0] - secondAnnotationsSorted[0][0]) *
  (firstAnnotationsSorted[0][1] - secondAnnotationsSorted[0][1])
)
```

Sorry, multiplication will not work but there should be a way to convert 0 to 1 and still got the result we want to. Kind of power or logarithm. Any ideas?

Reply:
I need to think about it more. I did it just blindly because this feature is not supported yet. Will do it when I work on it.

Reply:
You have a good point 🙂

Sorry, I just blindly read the code.